pyomo/opt/solver/shellcmd.py

```diff
@@ -116,7 +116,10 @@ def available(self, exception_flag=False):
             return True
         if not OptSolver.available(self,exception_flag):
             return False
-        ans = self.executable()
+        try:
+            ans = self.executable()
+        except NotImplementedError:
+            ans = None
```

Comment:
This looks problematic.  Suppressing an internal exception and replacing it with a different exception that doesn't contain the same information makes debugging *very* difficult.